test/Microsoft.AspNetCore.Mvc.ViewFeatures.Test/Internal/MemberExpressionCacheKeyComparerTest.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq.Expressions;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures
+{
+    public class MemberExpressionCacheKeyComparerTest
+    {
+        private readonly MemberExpressionCacheKeyComparer Comparer = MemberExpressionCacheKeyComparer.Instance;
+
+        [Fact]
+        public void Equals_ReturnsTrue_ForTheSameExpression()
+        {
+            // Arrange
+            var key = GetKey(m => m.Value);
+
+            // Act & Assert
+            VerifyEquals(key, key);
+        }
+
+        [Fact]
+        public void Equals_ReturnsTrue_ForDifferentInstances_OfSameExpression()
+        {
+            // Arrange
+            var key1 = GetKey(m => m.Value);
+            var key2 = GetKey(m => m.Value);
+
+            // Act & Assert
+            VerifyEquals(key1, key2);
+        }
+
+        [Fact]
+        public void Equals_ReturnsTrue_ForChainedMemberAccessExpressionsWithReferenceTypes()
```

Comment:
Not clear how `object` versus `struct` versus `Nullable<T>` matters to `MemberExpressionCacheKeyComparer`. Why do these variations exist?